### ISSUE TITLE
ManageConnection is not handling rate limiting

### ISSUE THREAD

**Issue by olovpe** [slack-go/slack#300]
ManageConnection is ignoring rate limit error when connecting and is retrying without waiting for `RetryAfter`.

```
2018-04-19 06:45:14.59 INFO (*Bot).handleEvents slack bot disconnected, intentional?  false
2018-04-19 06:45:14.59 INFO (*Bot).handleEvents Connecting attept: 1 count: 22
2018-04-19 06:45:15.10 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 32s
2018-04-19 06:45:15.21 INFO (*Bot).handleEvents Connecting attept: 2 count: 22
2018-04-19 06:45:15.50 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 32s
2018-04-19 06:45:15.64 INFO (*Bot).handleEvents Connecting attept: 3 count: 22
2018-04-19 06:45:16.12 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 31s
2018-04-19 06:45:16.35 INFO (*Bot).handleEvents Connecting attept: 4 count: 22
2018-04-19 06:45:16.75 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 30s
2018-04-19 06:45:17.06 INFO (*Bot).handleEvents Connecting attept: 5 count: 22
2018-04-19 06:45:17.36 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 30s
2018-04-19 06:45:18.35 INFO (*Bot).handleEvents Connecting attept: 6 count: 22
2018-04-19 06:45:18.85 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 28s
2018-04-19 06:45:19.99 INFO (*Bot).handleEvents Connecting attept: 7 count: 22
2018-04-19 06:45:20.43 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 27s
2018-04-19 06:45:21.82 INFO (*Bot).handleEvents Connecting attept: 8 count: 22
2018-04-19 06:45:22.10 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 25s
2018-04-19 06:45:26.14 INFO (*Bot).handleEvents Connecting attept: 9 count: 22
2018-04-19 06:45:26.44 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 21s
2018-04-19 06:45:36.66 INFO (*Bot).handleEvents Connecting attept: 10 count: 22
2018-04-19 06:45:37.03 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 10s
2018-04-19 06:46:14.35 INFO (*Bot).handleEvents Connecting attept: 11 count: 22
2018-04-19 06:46:15.41 INFO (*Bot).handleEvents slack bot connected
2018-04-19 06:46:15.41 INFO (*Bot).handleEvents slack bot disconnected, intentional?  false
2018-04-19 06:46:15.41 INFO (*Bot).handleEvents Connecting attept: 1 count: 23
2018-04-19 06:46:15.76 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 1m0s
2018-04-19 06:46:15.86 INFO (*Bot).handleEvents Connecting attept: 2 count: 23
2018-04-19 06:46:16.37 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 59s
2018-04-19 06:46:16.55 INFO (*Bot).handleEvents Connecting attept: 3 count: 23
2018-04-19 06:46:16.99 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 58s
2018-04-19 06:46:17.14 INFO (*Bot).handleEvents Connecting attept: 4 count: 23
2018-04-19 06:46:17.60 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 58s
2018-04-19 06:46:18.11 INFO (*Bot).handleEvents Connecting attept: 5 count: 23
2018-04-19 06:46:18.53 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 57s
2018-04-19 06:46:19.09 INFO (*Bot).handleEvents Connecting attept: 6 count: 23
2018-04-19 06:46:19.45 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 56s
2018-04-19 06:46:21.27 INFO (*Bot).handleEvents Connecting attept: 7 count: 23
2018-04-19 06:46:21.54 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 54s
2018-04-19 06:46:22.10 INFO (*Bot).handleEvents Connecting attept: 8 count: 23
2018-04-19 06:46:22.53 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 53s
2018-04-19 06:46:31.45 INFO (*Bot).handleEvents Connecting attept: 9 count: 23
2018-04-19 06:46:31.72 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 44s
2018-04-19 06:46:52.48 INFO (*Bot).handleEvents Connecting attept: 10 count: 23
2018-04-19 06:46:52.91 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 22s
2018-04-19 06:46:59.08 INFO (*Bot).handleEvents Connecting attept: 11 count: 23
2018-04-19 06:46:59.41 ERROR (*Bot).handleEvents connection error, post: Slack rate limit exceeded, retry after 16s
2018-04-19 06:47:31.20 INFO (*Bot).handleEvents Connecting attept: 12 count: 23
2018-04-19 06:47:32.95 INFO (*Bot).handleEvents slack bot connected
```

**Comment by james-lawrence** [comment]
yup, technically they come through `*RTM.IncomingEvents` as `*slack.ConnectionErrorEvent` IIRC.
definitely something we can improve for the managed connection. should be pretty easy to implement if you want to take a stab at it.

**Comment by armsnyder** [comment]
Hey @james-lawrence was this one resolved by 490f287f4b4412bf56d2d176d640cb916d008371?

**Comment by james-lawrence** [comment]
yup!
